### PR TITLE
Advanced sidecar custom

### PR DESCRIPTION
**Pull request by yvan33**

### CODE REVIEW

File: sidecar-advanced-custom/Notes_PostmanEnvironment.json [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+    },
+    {
+      "enabled": true,
+      "key": "rest_endpoint",
+      "value": "/rest/v10",
+      "type": "text"
+    },
+    {
+      "enabled": true,
+      "key": "Note_Path",
```
Comment (jcsmorais) [inline sidecar-advanced-custom/Notes_PostmanEnvironment.json]: This whole file will be removed from version control, only here for demonstration purposes while we're actively working on it! 

This is basically a copy of the environment settings partners will need to setup on Postman for any tutorial, the important and different piece required by our customization is the `Notes_Path` which they'll need to setup on their environment settings, which needs to match their local path to the [images folder](https://github.com/sugarcrm/uncon/pull/27/commits/0079d479cd30fffe8760f0952c50e34322d8ed7f).

File: sidecar-advanced-custom/custom/modules/Contacts/clients/base/views/yearbook/yearbook.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        }
+
+        notes.fetch(
+            {
+                fields: [
+                    'file_mime_type', 'file_ext','filename'
+                ],
+                params:
+                    {
+                        filter:[{'tag': {'$in': [tag.name]}}]
```
Comment (jcsmorais) [inline sidecar-advanced-custom/custom/modules/Contacts/clients/base/views/yearbook/yearbook.js]: Can we use all the tags in the record instead?
The current dataset we have makes use of two tags per record as such:

```
Mutants, Assassin 
Mutants, Genetically Enhanced
Mutants, Genius
Genius, Assassin
Human, Genius
```

File: sidecar-advanced-custom/Notes_PostmanEnvironment.json (hunk 1 above)
Comment (jcsmorais) [reply sidecar-advanced-custom/Notes_PostmanEnvironment.json]:
```
newman run Notes_PostmanCollection.json --environment Notes_PostmanEnvironment.json

File: sidecar-advanced-custom/README.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Creating a Yearbook in the Student's side pane
+
+## Prerequisites
+1. This tutorial requires to have the Professor M package intalled. Please fin the instructions [here](https://github.com/sugarcrm/uncon/tree/2017/ProfessorM)
```
Comment (jcsmorais) [inline sidecar-advanced-custom/README.md]: `find`?

File: sidecar-advanced-custom/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Creating a Yearbook in the Student's side pane
+
+## Prerequisites
+1. This tutorial requires to have the Professor M package intalled. Please find the instructions [here](https://github.com/sugarcrm/uncon/tree/2017/ProfessorM)
```
Comment (rhoggSugarcrm) [inline sidecar-advanced-custom/README.md]: Please change to:

This tutorial requires you to have the Professor M package installed.

File: sidecar-advanced-custom/README.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+2. Download `images.zip` and uncompress it on your machine.
+3. Open Postman, click on the gear icon in the upper right corner and select `Manage Environments`.
+4. Add a key called "Note_Path" and fill the value with the absolute path of your `images` folder. Exple: `/Users/<username>/Desktop/images`
```
Comment (rhoggSugarcrm) [inline sidecar-advanced-custom/README.md]: fill -> fill in

File: sidecar-advanced-custom/custom/modules/Contacts/clients/base/layouts/record/record.php [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+/*
+ * Your installation or use of this SugarCRM file is subject to the applicable
+ * terms available at
+ * http://support.sugarcrm.com/Resources/Master_Subscription_Agreements/.
+ * If you do not agree to all of the applicable terms or do not have the
+ * authority to bind the entity as an authorized representative, then do not
+ * install or use this SugarCRM file.
+ *
+ * Copyright (C) SugarCRM Inc. All rights reserved.
+ */
```
Comment (rhoggSugarcrm) [inline sidecar-advanced-custom/custom/modules/Contacts/clients/base/layouts/record/record.php]: Should we take out this license header?

File: sidecar-advanced-custom/custom/modules/Contacts/clients/base/views/record/record.php [hunk 7]
```diff
@@ -0,0 +1,342 @@
...
+/*
+ * Your installation or use of this SugarCRM file is subject to the applicable
+ * terms available at
+ * http://support.sugarcrm.com/Resources/Master_Subscription_Agreements/.
+ * If you do not agree to all of the applicable terms or do not have the
+ * authority to bind the entity as an authorized representative, then do not
+ * install or use this SugarCRM file.
+ *
+ * Copyright (C) SugarCRM Inc. All rights reserved.
+ */
```
Comment (rhoggSugarcrm) [inline sidecar-advanced-custom/custom/modules/Contacts/clients/base/views/record/record.php]: Ditto.

File: sidecar-advanced-custom/custom/modules/Contacts/clients/base/views/yearbook/yearbook.hbs [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<div class="btn-toolbar pull-right">
+    <div class="btn-group">
+        <button type="button" class="btn btn-invisible" data-action="close"><i class="fa fa-times"></i></button>
+    </div>
+</div>
+<div class="comic-pane title">
+    <div class="text">
+        <div class="holder">
+            <h3>
+                {{str 'LBL_YEARBOOK_HEADER' module}}
```
Comment (rhoggSugarcrm) [inline sidecar-advanced-custom/custom/modules/Contacts/clients/base/views/yearbook/yearbook.hbs]: We moved LBL_YEARBOOK_HEADER back into the app_strings, so this shouldn't have a `module` argument any more, right?

**Review by rhoggSugarcrm — CHANGES_REQUESTED** [review]
Looks good to me except one thing. @jcsmorais I can fix it if you agree with the change.

File: sidecar-advanced-custom/custom/modules/Contacts/clients/base/views/yearbook/yearbook.hbs (hunk 8 above)
Comment (jcsmorais) [reply sidecar-advanced-custom/custom/modules/Contacts/clients/base/views/yearbook/yearbook.hbs]: Right

**Review by rhoggSugarcrm — APPROVED** [review]

**Pull request merged**